zerver/decorator.py

```diff
@@ -582,7 +599,17 @@ def _wrapped_func_arguments(request: HttpRequest, *args: Any, **kwargs: Any) ->
             except Exception as err:
                 if is_webhook or webhook_client_name is not None:
                     request_body = request.POST.get('payload')
-                    if request_body is not None:
+                    # NOTE: importing this at the top of the file leads to a
+                    # cyclic import; correct fix is probably to move
+                    # UnexpectedWebhookEventType to a different module.
+                    from zerver.lib.webhooks.common import UnexpectedWebhookEventType
+                    if isinstance(err, UnexpectedWebhookEventType) and request_body is not None:
+                        log_exception_to_webhook_logger(
+                            request, profile,
+                            request_body=request_body,
+                            unexpected_event=True
+                        )
+                    elif request_body is not None:
                         log_exception_to_webhook_logger(request, profile,
                                                         request_body=request_body)
```

Comment:
This code would be cleaner with created a `kwargs` object that's empty unless it's an `UnexpectedWebhookEventType`; that way, you don't need to duplicate the call to `log_exception_to_webhook_logger`

Reply:
(You can put the check for exception type inside a higher-level `if request_body is not None` block).